### DISCUSSION TITLE
Apollo/client included in client-side bundle

### DISCUSSION THREAD

**Discussion by mckennapaul27** [vercel/next.js#37569] [Help] [answered]
I am using Nextjs (SSG) + Strapi + Apollo/client for a website and everything is working perfectly. However I am trying to reduce client JS bundle size on Nextjs and apollo/client is the biggest chunk that I am struggling to reduce/remove/split.

All my pages are SSG and therefore since apollo/client is only used within getStaticProps, I believed that apollo/client should not be bundled for the client side.

By a long process of elimination, I can see that even when I remove all references to apollo/client within /pages folder (by commenting out all the getStaticProps functions) it is still bundled.

Simply having the following code within `lib/server-api.js` even without using this code anywhere in the app means apollo is bundled for the client-side:

```
import {
ApolloClient,
InMemoryCache,
gql,
HttpLink,
concat,
ApolloLink,
} from ‘@apollo/client’;

const httpLink = new HttpLink({
uri: `${
process.env.NODE_ENV === ‘development’
? ‘http://localhost:1337’
: process.env.NEXT_PUBLIC_STRAPI_API_URL
}/graphql`,
});

const authMiddleware = new ApolloLink((operation, forward) => {
operation.setContext(({ headers = {} }) => ({
headers: {
…headers,
authorization: `Bearer ${
process.env.NODE_ENV === ‘development’
? process.env.API_TOKEN_LOCAL
: process.env.API_TOKEN
}`,
},
}));

return forward(operation);
});
const client = new ApolloClient({
cache: new InMemoryCache(),
link: concat(authMiddleware, httpLink),
});
```

When I comment out the above, it is no longer bundled. But as you can see from above, this code is located in `lib/server-api.js` and nothing is exported.

So why is it bundling to the client-side?

Any help greatly appreciated!!

**Comment by icyJoseph** [comment, marked as answer]
Hi

I copied your file and put it on a project of mine, installed `@apollo/client` and run the [bundle analyzer](https://www.npmjs.com/package/@next/bundle-analyzer), and found zero traces of the `@apollo/client` on the client side bundle.

Are you sure you don't import from `@apollo/client` elsewhere?

I am also surprised because next does [code-elimination](https://next-code-elimination.vercel.app/), to drop various artifacts from the client if these are not needed.

Do you have a repo we could take a look at?

**Reply by mckennapaul27** [reply]
Hey @icyJoseph  thanks for your reply, really appreciate it.

After further digging I found that this was caused by importing a function from `@/lib/server-api` that was not used in getStaticProps:

```
import {
    getBlogHome, // getBlogHome not used in getStaticProps
    getGlobalData,
    getMenuItemsByLocale,
    getPaginatedBlogs,
} from '@/lib/server-api';
```

When I removed `getBlogHome`, it resolved the issue!

Guess I need to improve my linting :-)

Thanks for the advice.

**Reply by icyJoseph** [reply]
Yes, or in this situation this tool also helps a bunch https://next-code-elimination.vercel.app/
